l1-contracts/contracts/bridge/ntv/NativeTokenVault.sol

```diff
@@ -0,0 +1,419 @@
+        uint256 balanceBefore = _token.balanceOf(address(this));
+        // slither-disable-next-line arbitrary-send-erc20
+        _token.safeTransferFrom(_from, address(this), _amount);
+        uint256 balanceAfter = _token.balanceOf(address(this));
+
+        return balanceAfter - balanceBefore;
+    }
+
+    /// @param _token The address of token of interest.
+    /// @dev Receives and parses (name, symbol, decimals) from the token contract
+    function getERC20Getters(address _token) public view override returns (bytes memory) {
+        return BridgeHelper.getERC20Getters(_token, ETH_TOKEN_ADDRESS);
+    }
+
+    /// @notice Returns the parsed assetId.
+    /// @param _nativeToken The address of the token to be parsed.
+    /// @dev Shows the assetId for a given chain and token address
+    function getAssetId(uint256 _chainId, address _nativeToken) external pure override returns (bytes32) {
+        return DataEncoding.encodeNTVAssetId(_chainId, _nativeToken);
+    }
+
+    /*//////////////////////////////////////////////////////////////
+                            TOKEN DEPLOYER FUNCTIONS
+    //////////////////////////////////////////////////////////////*/
+
+    function _ensureTokenDeployed(
+        uint256 _originChainId,
+        bytes32 _assetId,
+        address _originToken,
+        bytes memory _erc20Data
+    ) internal virtual returns (address _token) {
+        address expectedToken = _assetIdCheck(_originChainId, _assetId, _originToken);
+        _token = _ensureTokenDeployedInner({
+            _originChainId: _originChainId,
+            _assetId: _assetId,
+            _originToken: _originToken,
+            _erc20Data: _erc20Data,
+            _expectedToken: expectedToken
+        });
+    }
+
+    function _assetIdCheck(
+        uint256 _originChainId,
+        bytes32 _assetId,
+        address _originToken
+    ) internal view returns (address expectedToken) {
+        expectedToken = calculateCreate2TokenAddress(_originChainId, _originToken);
+        bytes32 expectedAssetId = DataEncoding.encodeNTVAssetId(_originChainId, _originToken);
+        if (_assetId != expectedAssetId) {
+            // Make sure that a NativeTokenVault sent the message
+            revert AssetIdMismatch(_assetId, expectedAssetId);
+        }
+    }
+
+    function _ensureTokenDeployedInner(
+        uint256 _originChainId,
+        bytes32 _assetId,
+        address _originToken,
+        bytes memory _erc20Data,
+        address _expectedToken
+    ) internal returns (address) {
+        address deployedToken = _deployBridgedToken(_originChainId, _originToken, _erc20Data);
+        if (deployedToken != _expectedToken) {
+            revert AddressMismatch(_expectedToken, deployedToken);
+        }
+
+        isTokenBridged[_assetId] = true;
+        tokenAddress[_assetId] = _expectedToken;
+        return _expectedToken;
```

Comment:
do you think this function should return any value considering it is always the same value that was passed into it?